brain/s3/utils/formatters.py

```
@@ -0,0 +1,58 @@
+    lambda agent, obs, response: _attempt_code_creation(
+        agent, parse_code_from_string(response), obs
+    )
+    is not None
+)
+code_valid_error_msg = "Incorrect code: The agent action must be a valid function and use valid parameters from the docstring list."
+CODE_VALID_FORMATTER = lambda agent, obs, response: (
+    code_valid_check(agent, obs, response),
+    code_valid_error_msg,
+)
+
+thoughts_answer_tag_check = lambda response: split_thinking_response(response)[1] != ""
```

Ruff (0.14.6)

Do not assign a `lambda` expression, use a `def`

Rewrite `thoughts_answer_tag_check` as a `def`

(E731)